src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,124 @@
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.UserIdMismatchException;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+/**
+ * class controller of one user
+ */
+@RestController
+@RequestMapping("/api/user")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    /**
+     * Method that find all the information of user entity
+     * @return: a collection type list with the information of user table
+     */
+    @GetMapping
+    public Iterable findAll() {
+        return userRepository.findAll();
+    }
+
+    /**
+     * Method that find information by primary key
+     * @param id : primary key of one table
+     * @return: the information of with  id you send.
+     */
+    @GetMapping("/{id}")
+    public User findOne(@PathVariable Long id) throws UserNotFoundException {
+        return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(
+                ErrorConstants.NOT_EXIST_USER_ID));
+    }
+
+    /**
+     * Method that save the fields from an entity.
+     * @param user object of entity user
+     * @return: null(message of success or warning )
+     */
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    /**
+     * Method that delete a registry of one entity by primary key
+     * @param id :primary key of one table
+     */
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(
+                ErrorConstants.NOT_EXIST_USER_ID));
+        userRepository.deleteById(id);
+    }
+
+    /**
+     * method that update a registry by primary key
+     * @param user : object of entity user
+     * @param id : primary key of one table
+     * @return null(message of success or warning )
+     */
+    @PutMapping("/{id}")
+    public User updateBook(@RequestBody User user, @PathVariable Long id) throws UserIdMismatchException {
+        if (user.getId() != id) {
+            throw new UserIdMismatchException();
+        }
+        userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(
+                ErrorConstants.NOT_EXIST_USER_ID));
+        return userRepository.save(user);
+    }
+    /**
+     * method that save a registry a book
+     * @param id: user_id from user table
+     * @param book: object book to save
+     * @return null(message of success or warning )
+     * @throws UserNotFoundException
+     */
+    @PutMapping("/{id}/books")
+    @ResponseStatus(HttpStatus.CREATED)
+    public User addBook(@PathVariable Long id, @RequestBody Book book)
```

Comment:
The method create is with post, it isn't put.
Also, it is not necessary to pass the id by params, since all the information of the book is in the body

Reply:
I changed the verb and I removed the param